Add household and dwelling unit identifier to instrument enumerator

jdzak opened the pull request:
Added ability for the instrument enumerator to map household and dwelling unit ids.  It uses the public_id from the the highest ranking household or dwelling unit.

For bug details see..
https://code.bioinformatics.northwestern.edu/issues/issues/show/4121

Commit:
Add du_id and hh_id to instrument enumerator #4121

Commit:
Remove debugging code. #4121

Commit:
Use household person links to find household unit. #4121

Commit:
Fix order by rank method. #4121

Commit:
Add method to sort dwelling household links by rank. #4121

Commit:
Enable household unit id example.

This required switching the survey that populates the pre_preg to spec_blood table because spec_blood has an hh_id property.

Commit:
Simplify ordering code slightly. #4121

Commit:
Methods no longer used. #4121

Commit:
DRY up code dealing with COMMUNICATION_RANK_CL1. #4121

jdzak commented on app/models/communication_rank_cl_one.rb:
```diff
@@ -0,0 +1,15 @@
+# Helper methods for dealing with the COMMUNICATION_RANK_CL1 code list
+#
+module CommunicationRankCLOne
+  ORDER = [1, 2, -5, 3, 4, -4].map do |rank|
```
My interpretation of the ordering (1=primary, 2=secondary, -5=other, 3=invalid, 4=duplicate, -4=missing) made the most sense to me. If anyone else has an alternative, I'm open to discussion.

flockom replied:
Had the same discussion about {ADDRESS,EMAIL,PHONE}_RANK and decided to keep them in MDES order

https://github.com/NUBIC/ncs_navigator_core/commit/e4272caa3563098dac3b863fdb50637edcd560a6#L0R10

jdzak commented on spec/lib/ncs_navigator/core/warehouse/instrument_to_warehouse_spec.rb:
```diff
@@ -50,14 +50,11 @@ def create_response_for(question, response_set=response_set)
-          a_5 "Poor?"
+          :data_export_identifier=>"SPEC_BLOOD.HEMOPHILIA"
+          a_1 "Yes"
+          a_2 "No"
```
Switching the instrument from pre_preg to spec_blood allows us to test all the same functionality and then also the du_id and hh_id population.

Commit:
Expect situation when rank doesn't exist. #4121

Commit:
Switching to mdes ordering. #4121

The codes seems intentionally ordered, so even though I don't agree with the ordering I will use the order the MDES prescribes.
